src/main/java/com/metamx/cache/CaffeineCache.java

```diff
@@ -54,58 +55,79 @@ public static CaffeineCache create(final CaffeineCacheConfig config)
           .expireAfterAccess(config.getExpiration(), TimeUnit.MILLISECONDS);
     }
     if (config.getMaxSize() >= 0) {
-      builder = builder
-          .maximumSize(config.getMaxSize())
-          .maximumWeight(config.getMaxSize());
+      builder = builder.maximumWeight(config.getMaxSize());
+      builder.weigher(new Weigher<NamedKey, byte[]>()
+      {
+        @Override
+        public int weigh(@Nonnull NamedKey key, @Nonnull byte[] value)
+        {
+          return value.length;
+        }
+      });
+    }
+
+    final Cache delegateCache = config.getDelegateCache();
+    final CacheLoader<NamedKey, byte[]> cacheLoader;
+    if (delegateCache != null) {
+      cacheLoader = new CacheLoader<NamedKey, byte[]>()
+      {
+        @Override
+        public byte[] load(@Nonnull NamedKey key) throws Exception
+        {
+          return delegateCache.get(key);
+        }
+
+        @Override
+        public Map<NamedKey, byte[]> loadAll(@Nonnull Iterable<? extends NamedKey> keys) throws Exception
+        {
+          final Iterable<NamedKey> keys0 = (Iterable<NamedKey>) keys;
+          return Maps.transformValues(delegateCache.getBulk(keys0), CaffeineCache::serialize);
```

Comment:
Thanks @ben-manes 

Note this PR is an experiment I did right before heading off on PTO for a bit.

In Druid there was a PR a bit ago that allows a L1/L2 style cache: https://github.com/druid-io/druid/pull/1800 

This allowed having a local fast-but-small cache that was backed by something like memcached.

This PR is a basic investigation to see how well caffeine functions as a substitute for the L1/L2 style caching PR mentioned above.